xdsl/dialects/arm_neon.py

```diff
@@ -106,7 +113,7 @@ class VectorWithArrangement(AssemblyInstructionArg):
 
     def __init__(
         self,
-        reg: NEONRegisterType,
+        reg: NEONRegisterType | SSAValue,
```

Comment:
My feeling is that `self.reg` should be a NEONRegisterType, and that the init should do a `if isinstance(reg, SSAValue): assert isinstance(reg.type, NEONRegisterType); reg = reg.type`

Reply:
updated with these changes. I'm still getting red squiggly line warnings in the ide but the tests are passing

Reply:
ok the pyright checks are catching it in CI, not surprising I suppose